### PR TITLE
perf(table): only issue 1 iterator for full range scan

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

When scanning for a table, if the range is unbounded and order is not required, we can create a single iterator for each continuous vnode range. This can be helpful for plain `Chain` and batch scans without order requirements (if we support specifying it as mentioned in #6925).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #5594

### CODE REVIEW

**Review by jon-chuang — COMMENTED** [review]
LGTM

File: src/storage/src/table/batch_table/storage_table.rs [hunk 1]
```diff
@@ -295,21 +296,38 @@ impl<S: StateStore> StorageTable<S> {
...
+            // for each continuous vnode range.
+
+            // In this case, the `vnode_hint` must be default for singletons and `None` for
+            // distributed tables.
+            assert_eq!(vnode_hint.unwrap_or(DEFAULT_VNODE), DEFAULT_VNODE);
+
+            Either::Left(self.vnodes.high_ranges().map(|r| {
+                let start = Included(VirtualNode::from_index(*r.start()).to_be_bytes().to_vec());
+                let end = end_bound_of_prefix(&VirtualNode::from_index(*r.end()).to_be_bytes());
+                assert_matches!(end, Excluded(_) | Unbounded);
```
Comment (chenzl25) [inline src/storage/src/table/batch_table/storage_table.rs:313]: Just to confirm when the last `vnode` = 255, `end` will be `Unbounded` ?

**Review by chenzl25 — APPROVED** [review]
LGTM!

File: src/storage/src/table/batch_table/storage_table.rs (hunk 1 above)
Comment (BugenZhao) [reply src/storage/src/table/batch_table/storage_table.rs:313]: Yes. The `end_bound` will be `Unbounded` if all bytes in the given prefix is '\xff'.

**Pull request merged**